gateway/session_manager.go

```diff
@@ -418,3 +442,33 @@ func GetAccessDefinitionByAPIIDOrSession(session *user.SessionState, api *APISpe
 
 	return accessDef, allowanceScope, nil
 }
+
+type endpointRateLimitInfo struct {
+	keySuffix string
+	rate      int64
+	per       int64
+}
+
+func getEndpointRateLimitInfo(method string, path string, endpoints []user.Endpoint) (*endpointRateLimitInfo, bool) {
```

Comment:
this can remain here just return a *model.EndpointRateLimitInfo

Reply:
this struct is used in place, not used elsewhere, so I feel it's the best to be defined here unless it gets usage across multiple packages.